digits/utils/forms.py

```diff
@@ -339,26 +339,39 @@ class MultiNumberRange(object):
         interpolated using `%(min)s` and `%(max)s` if desired. Useful defaults
         are provided depending on the existence of min and max.
     """
-    def __init__(self, min=None, max=None, message=None):
+    def __init__(self, min=None, max=None, min_inclusive=True, max_inclusive=True, message=None):
         self.min = min
         self.max = max
         self.message = message
+        self.min_inclusive = min_inclusive
+        self.max_inclusive = max_inclusive
 
     def __call__(self, form, field):
         fdata = field.data if isinstance(field.data, (list, tuple)) else [field.data]
         for data in fdata:
-            if data is None or (self.min is not None and data < self.min) or \
-                    (self.max is not None and data > self.max):
+            print 'checking', data, self.min
+            flags = 0
+            flags |= (data is None) << 0
+            flags |= (self.min is not None and self.min_inclusive and data < self.min) << 1
+            flags |= (self.max is not None and self.max_inclusive and data > self.max) << 2
+            flags |= (self.min is not None and not self.min_inclusive and data <= self.min) << 3
+            flags |= (self.max is not None and not self.max_inclusive and data >= self.max) << 4
+
```

Comment:
I had borrowed if from the wtforms case where there is only a min and max, and it has logic for cases min!=None, max!=None, and (min!=None && max!=None).  I had initially started working on a message such that there was one message that handled any permutation of settings, that's when I added the flags, and decided against it.  I don't find the flags offensive, but would be happy to change them if it's confusing or messy.